python/pyarrow/table.pxi

```diff
@@ -159,7 +159,7 @@ cdef class Column:
         sp_column.reset(new CColumn(boxed_field.sp_field, arr.sp_array))
         return pyarrow_wrap_column(sp_column)
 
-    def to_pandas(self, strings_to_categorical=False):
+    def to_pandas(self, strings_to_categorical=False, zero_copy_only=False):
```

Comment:
I would guess that this nearly always fails to the BlockManager. It will work for Arrays/Series but as Pandas will always allocate a large matrix for columns of the same type, you most likely get near to no zero-copies. (e.g. a DataFrame with two float64 columns will need a copy.)

Reply:
@xhochy That makes sense - but why wouldn't you just make one `PandasBlock` per column (without copying), instead of created a `PandasBlock` per type and copying the various columns of that type into it? 

Also, I've pushed a fix for the AppVeyor warning but the Travis errors seem unrelated (in the Go & JS codebases)?

Reply:
@njwhite Because that is the underlying assumption based upon which Pandas 0.x DataFrames work on. There are several functions that use this assumptions to provide certain (slicing) features. @wesm might be able to go into detail/be more concrete.

Reply:
@xhochy Makes sense, definitely want to keep this change focussed so should I just remove the flag from `pa.Table`? I'll fix the AppVeyor complaints tonight, & rebase off master to see if that fixes the other issues.

Reply:
No, keep it. I just wanted to give you the hint that it will likely fail to do zero-copy in >90% of all cases.